frontend/src/pages/accounts/profile.test.tsx

```diff
@@ -449,6 +450,151 @@ describe("The dashboard", () => {
+    setMockRuntimeDataOnce(getMockRuntimeDataWithPeriodicalPremium());
+    const mockedConfig = mockConfigModule.getRuntimeConfig();
+    // getRuntimeConfig() is called frequently, so mock its return value,
+    // then restore the original mock at the end of this test:
+    mockConfigModule.getRuntimeConfig.mockReturnValue({
+      ...mockedConfig,
+      maxFreeAliases: 1,
+    });
+
+    render(<Profile />);
+
+    const bannerHeaderNonUS = screen.getByText(
+      "l10n string: [profile-maxed-aliases-without-phone-header], with vars: {}",
+    );
+
+    const bannerDescriptionNonUS = screen.getByText(
+      `l10n string: [profile-maxed-aliases-without-phone-description], with vars: {"limit":${
+        mockConfigModule.getRuntimeConfig().maxFreeAliases
+      }}`,
+    );
+
+    expect(bannerHeaderNonUS).toBeInTheDocument();
+    expect(bannerDescriptionNonUS).toBeInTheDocument();
+
+    mockConfigModule.getRuntimeConfig.mockReturnValue(mockedConfig);
+  });
+
+  it("shows tooltip when user hovers over the number of masks when user reach the limit", async () => {
+    setMockAliasesDataOnce({ random: [{ enabled: true, id: 42 }], custom: [] });
+    setMockProfileDataOnce({ has_premium: false });
+    setMockRuntimeDataOnce(getMockRuntimeDataWithPeriodicalPremium());
+    const mockedConfig = mockConfigModule.getRuntimeConfig();
+    // getRuntimeConfig() is called frequently, so mock its return value,
+    // then restore the original mock at the end of this test:
+    mockConfigModule.getRuntimeConfig.mockReturnValue({
+      ...mockedConfig,
+      maxFreeAliases: 1,
+    });
+
+    render(<Profile />);
+
+    const totalAliases = screen.getByText(/profile-stat-label-aliases-used-2/);
+
+    expect(
+      screen.queryByText(/profile-maxed-aliases-tooltip/),
+    ).not.toBeInTheDocument();
+
+    // eslint-disable-next-line testing-library/no-node-access
+    const parent = totalAliases.parentElement as HTMLElement;
+    const trigger = within(parent).getByText(
+      `${mockConfigModule.getRuntimeConfig().maxFreeAliases}`,
+    );
+
+    await userEvent.hover(trigger);
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(/profile-maxed-aliases-tooltip/),
+      ).toBeInTheDocument();
+    });
+
+    mockConfigModule.getRuntimeConfig.mockReturnValue(mockedConfig);
+  }, 10000);
+
+  it("Upgrade to premium CTA takes user to premium price plans", async () => {
+    setMockAliasesDataOnce({ random: [{ enabled: true, id: 42 }], custom: [] });
+    setMockProfileDataOnce({ has_premium: false });
+    setMockRuntimeDataOnce(getMockRuntimeDataWithPeriodicalPremium());
+    const mockedConfig = mockConfigModule.getRuntimeConfig();
+    // getRuntimeConfig() is called frequently, so mock its return value,
+    // then restore the original mock at the end of this test:
+    mockConfigModule.getRuntimeConfig.mockReturnValue({
+      ...mockedConfig,
+      maxFreeAliases: 1,
+    });
```

Comment:
suggestion (non-blocking): This is the 4th copy-paste of this test setup code. Refactor it into a separate function? Either here in this test, or maybe "up" in a module in `frontend/__mocks__/hooks/api` somewhere?

(Note: [the other test setup](https://github.com/mozilla/fx-private-relay/pull/4038/files#diff-eb5b70afbfb6260a42fd6afd04933e7c4d95ef09d2dd67907dd11e08e0947360R478) calls `getMockRuntimeDataWithPhones` so it still needs its own custom setup code.

Reply:
done